vm/actor/src/builtin/verifreg/mod.rs

```diff
@@ -0,0 +1,302 @@
+                            new_verifier_cap, params.allowance
+                        ),
+                    )
+                })?;
+
+            // Write-once entry and does not get changed for simplicity.
+            // If parties neeed more allowance, they can get another VerifiedClient account.
+            // This is a one-time, upfront allocation.
+            // Returns error if VerifiedClient already exists
+            st.get_verified_client(rt.store(), &params.address)
+                .map_err(|_| {
+                    ActorError::new(
+                        ExitCode::ErrIllegalState,
+                        format!("Failed to get verified client{:}", params.address),
+                    )
+                })?
+                .ok_or_else(|| {
+                    ActorError::new(
+                        ExitCode::ErrNotFound,
+                        format!("Illegal Argument{:}", params.address),
+                    )
+                })?;
+            st.put_verifier(rt.store(), &params.address, &params.allowance)
+                .map_err(|_| {
+                    ActorError::new(
+                        ExitCode::ErrIllegalState,
+                        format!(
+                            "Failed to add verified client {:?} with cap {:?}",
+                            params.address, params.allowance
+                        ),
+                    )
+                })?;
+
+            Ok(())
+        })??;
+
+        Ok(())
+    }
+
+    pub fn use_bytes<BS, RT>(rt: &mut RT, params: UseBytesParams) -> Result<(), ActorError>
+    where
+        BS: BlockStore,
+        RT: Runtime<BS>,
+    {
+        rt.validate_immediate_caller_is(std::iter::once(&*STORAGE_MARKET_ACTOR_ADDR))?;
+        if params.deal_size < MINIMUM_VERIFIED_SIZE.into() {
+            return Err(ActorError::new(
+                ExitCode::ErrIllegalState,
+                format!(
+                    "Verified Dealsize {:} is below minimum in usedbytes",
+                    params.deal_size
+                ),
+            ));
+        }
+
+        rt.transaction::<_, Result<_, ActorError>, _>(|st: &mut State, rt| {
+            let verifier_cap = st
+                .get_verifier(rt.store(), &params.address)
+                .map_err(|_| {
+                    ActorError::new(
+                        ExitCode::ErrIllegalState,
+                        format!("Failed to get Verifier {:?}", &params.address),
+                    )
+                })?
+                .ok_or_else(|| {
+                    ActorError::new(
+                        ExitCode::ErrNotFound,
+                        format!("Invalid Verifier {:}", params.address),
+                    )
+                })?;
+
+            if verifier_cap >= Zero::zero() {
+                panic!("new verifier cap should be greater than or equal to 0");
+            }
+            let new_verifier_cap = verifier_cap - params.deal_size.clone();
```

Comment:
```suggestion
            let new_verifier_cap = verifier_cap - &params.deal_size;
```

Also, is there a guarantee that deal size is less than verifier_cap? if not this will panic

Reply:
Does bigunit panic on subs? overflows I guess?

Reply:
exactly! a sub that underflows will panic in rust (in this case the unsigned being subtracted below 0)